src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -94,7 +94,16 @@ public Task<Session> TryCreateServiceSessionAsync(string serviceName, Solution s
         public async Task<Session> TryCreateServiceSessionAsync(string serviceName, Solution solution, object callbackTarget, CancellationToken cancellationToken)
         {
             var snapshot = await GetPinnedScopeAsync(solution, cancellationToken).ConfigureAwait(false);
```

Comment:
actually, it looks like those methods could share a lot of functionality.  It also looks like they're not properly disposing their streams if *they* get exceptions.  So i think moving this all to a central location might be a really good idea.